### PR TITLE
fix(Gregorian to JDC): handle Gregorian dates prior to 15.10.1582 correctly

### PR DESCRIPTION
**Pull request by tobiasschweizer**
This PR adds assertions to detect a bug in https://github.com/dhlab-basel/JDNConvertibleCalendar/blob/51331f72fb9ed53dbba75bafab20580509b5fe27/src/JDNCalendarConversion.ts#L50

### CODE REVIEW

**Comment by tobiasschweizer** [comment]
@gautschr I am getting three failing tests:


  1) Conversion of a Gregorian calendar date to JDC
       convert the Gregorian Calendar date 01-02(-1899) to JDC:

      AssertionError [ERR_ASSERTION]: JDC is wrong: received JDN is 1027496.5 instead of 1027495.5
      + expected - actual

      -1027496.5
      +1027495.5
      
      at checkJDC (dist/test/conversions.js:60:12)
      at Context.<anonymous> (dist/test/conversions.js:99:9)
      at processImmediate (internal/timers.js:439:21)
      at process.topLevelDomainCallback (domain.js:131:23)

  2) Conversion of a Gregorian calendar date to JDC
       convert the Gregorian Calendar date 28-02(-99) to JDC:

      AssertionError [ERR_ASSERTION]: JDC is wrong: received JDN is 1684959.5 instead of 1684958.5
      + expected - actual

      -1684959.5
      +1684958.5
      
      at checkJDC (dist/test/conversions.js:60:12)
      at Context.<anonymous> (dist/test/conversions.js:109:9)
      at processImmediate (internal/timers.js:439:21)
      at process.topLevelDomainCallback (domain.js:131:23)

  3) Get the number of days for a given month
       create a Gregorian date and get the number of days for a given BCE Gregorian date's month :

      AssertionError [ERR_ASSERTION]: wrong number of days
      + expected - actual

      -27
      +28
      
      at Context.<anonymous> (dist/test/conversions.js:1048:16)
      at processImmediate (internal/timers.js:439:21)
      at process.topLevelDomainCallback (domain.js:131:23)

File: src/JDNCalendarConversion.ts [hunk 1]
```diff
@@ -68,20 +66,19 @@ export module JDNConvertibleConversionModule {
...
-        if (year < 0) {
-            c = -0.75;
-        }
-        //
-        // we enforce Gregorian calendar for *all* dates (also prior to 1582)
-        //
+        let b = 0;
         let a = truncateDecimals(year/100.);
-        let b = 2 - a + truncateDecimals(a/4.);
+        let idate = year*10000 + month*100 + day;
```
Comment (tobiasschweizer) [inline src/JDNCalendarConversion.ts:71]: @gautschr Could you add a comment explaining why it is important to differentiate between dates before and after (or equal) 15.10.1582?

File: test/conversions.ts [hunk 2]
```diff
@@ -92,7 +92,7 @@ describe('Conversion of a Gregorian calendar date to JDC', () => {
         const gregorianCalendarDate: CalendarDate = new CalendarDate(1582, 10, 4, undefined, 0);
         const jdc: number = JDNConvertibleConversionModule.gregorianToJDC(gregorianCalendarDate);
 
-        checkJDC(2299149.5, jdc);
+        checkJDC(2299159.5, jdc);
     });
 
     it('convert the Gregorian Calendar date 15-10-1582 to JDC', () => {
```
Comment (tobiasschweizer) [inline test/conversions.ts:98]: the comment `15-10-1582` does not fit `new CalendarDate(-1001, 12, 22, undefined, 0)`. I guess this was always wrong, I'll fix it.

File: src/JDNCalendarConversion.ts (hunk 1 above)
Comment (tobiasschweizer) [reply src/JDNCalendarConversion.ts:71]: maybe there could be a remark in the README about this as well.

**Review by gautschr — APPROVED** [review]

**Pull request merged**